### DISCUSSION TITLE
Adding direction property

### DISCUSSION THREAD

**Discussion by Waseem-Almoliky** [tailwindlabs/tailwindcss#7258] [Ideas]
simply adding 2 classes like
`dir-ltr` and `dir-rtl`
or
`direction-ltr` and `direction-rtl`
to control the direction property
[https://developer.mozilla.org/en-US/docs/Web/CSS/direction](https://developer.mozilla.org/en-US/docs/Web/CSS/direction)

**Comment by NovusOpus** [comment]
Tailwind CSS Already support [RTL (Docs Link)](https://tailwindcss.com/docs/hover-focus-and-other-states#rtl-support) it's currently experimental since it may change in the future based on the community feedback.

**Reply by Waseem-Almoliky** [reply]
@NovusOpus I appreciate your response and answer.
![image](https://user-images.githubusercontent.com/57758482/155875768-43b2210a-c143-483f-97d1-8e80e751f13c.png)

what I meant is to add the CSS property `direction: rtl;` and  `direction: ltr;` as classes.

**Comment by rparmentier** [comment]
Upvoting this in 2023. I'm baffled this doesn't exists yet.

**Reply by adamwathan** [reply]
You can use any CSS property you want in Tailwind with arbitrary properties like `[direction:rtl]`, but we don't include `direction` in core because its use is discouraged by MDN:

<img width="824" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/4323180/14412c33-322a-40c8-a428-b99a2858b555">

**Reply by Waseem-Almoliky** [reply]
It may not be useful for majority but it's essential for "RTL" websites to show emails, Phone numbers, money and various things always by "LTR".
I always add it as a custom class when I work with Arabic as example.

**Reply by mohamadaliakbari** [reply]
Still It's necessary to have ltr class to style inputs such as email address, mobile number, credit carts and ...
